Dockerfile

```diff
@@ -90,6 +90,25 @@ ENV PIP_FIND_LINKS=https://storage.googleapis.com/jax-releases/libtpu_releases.h
 RUN pip install .[tpu]
 COPY . .
 
+################################################################################
+# GPU container spec.                                                          #
+################################################################################
+
+FROM base AS gpu
+
+
+RUN apt-get install -y google-perftools
+
+RUN JAX_VERSION=$(grep "jax==" pyproject.toml | sed -E 's/"jax==([0-9.]+)",/\1/') && \
+    pip install --no-cache-dir --upgrade \
+      "jax[cuda12_pip]==${JAX_VERSION}" \
+      -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
```

Comment:
Could this be in pyproject too, since we already specify `PIP_FIND_LINKS` below?

Reply:
fixed